crates/bevy_asset/src/io/mod.rs

```diff
@@ -248,10 +248,7 @@ impl AsyncRead for VecReader {
 /// Appends `.meta` to the given path.
 pub(crate) fn get_meta_path(path: &Path) -> PathBuf {
     let mut meta_path = path.to_path_buf();
-    let mut extension = path
-        .extension()
-        .expect("asset paths must have extensions")
-        .to_os_string();
+    let mut extension = path.extension().unwrap_or_default().to_os_string();
     extension.push(".meta");
     meta_path.set_extension(extension);
```

Comment:
nit: Could this be replaced with `meta_path.as_mut_os_string().push(b".meta");` now? Fishing out the extension seems like unnecessary work. (Would mean MSRV 1.70 though.)

Reply:
It could be, but I'm going to ignore since I'm uncertain what the status on the MSRV is right now, and I'd rather avoid the churn.

Reply:
Bevy's MSRV policy is always "latest stable", but if things fail you'll want to bump the MSRV in Cargo.toml. Totally fine to ignore this for now.